src/adaptors/in-memory-articles.ts

```diff
@@ -1,43 +1,45 @@
-import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+import { JsonLdObj } from 'jsonld/jsonld-spec';
+import { BlankNode, Term } from 'rdf-js';
 import { schema } from 'rdf-namespaces';
+import { stringToTerm, termToString } from 'rdf-string';
 import Articles from '../articles';
 import ArticleNotFound from '../errors/article-not-found';
 import NotAnArticle from '../errors/not-an-article';
 
 export default class InMemoryArticles implements Articles {
-  private articles: { [key: string]: [Iri, JsonLdObj] } = {};
+  private articles: { [id: string]: [BlankNode, JsonLdObj] } = {};
 
-  async set(id: Iri, article: JsonLdObj): Promise<void> {
+  async set(id: BlankNode, article: JsonLdObj): Promise<void> {
     const types = [].concat(article['@type'] || []);
 
-    if (!(types.includes(schema.Article)) || article['@id'] !== id) {
-      throw new NotAnArticle(types);
+    if (!(types.includes(schema.Article)) || article['@id'] !== termToString(id)) {
+      throw new NotAnArticle(types.map((type: string): Term => stringToTerm(type)));
```

Comment:
weird conversion due to the fact that `types` is not a RDF/JS object, but I guess it could go away later when the library is used more extensively

Reply:
Yeah, it will.